frontend/src/lib/components/SelectBox.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useRef, useEffect, useState } from 'react'
+import { kea, useActions, useValues } from 'kea'
+import { Col, Row, Input, Divider } from 'antd'
+import { actionFilterDropdownLogicType } from 'types/scenes/insights/ActionFilter/ActionFilterDropdownType'
+import { List } from 'antd'
+import { DownOutlined, RightOutlined } from '@ant-design/icons'
+import { EntityTypes } from '../../scenes/insights/trendsLogic'
+import Fuse from 'fuse.js'
+import { ActionType } from '~/types'
+
+export interface SelectBoxItem {
+    dataSource: SelectedItem[]
+    renderInfo({ item }: { item: SelectedItem }): JSX.Element
+    name: JSX.Element | string
+}
+
+export interface SelectedItem {
+    name: string
+    key: string
+    action?: ActionType
+    event?: string
+    volume?: number
+    usage_count?: number
+    category?: string
+}
+
+const scrollUpIntoView = (key: string): void => {
+    const searchList = document.querySelector('.search-list')
+    const item = document.querySelector('.action-filter-dropdown [datakey="' + key + '"]')
+    const diff = item?.getBoundingClientRect().top - searchList?.getBoundingClientRect().top
+    if (diff - 30 < 0) searchList.scrollTop = searchList.scrollTop + diff - 30
+}
+const scrollDownIntoView = (key: string): void => {
+    const searchList = document.querySelector('.search-list')
+    const item = document.querySelector('.action-filter-dropdown [datakey="' + key + '"]')
+    const diff = item?.getBoundingClientRect().top - searchList?.getBoundingClientRect().bottom
+    if (diff + 30 > 0) searchList.scrollTop = searchList.scrollTop + diff + 30
+}
+
+export const actionFilterDropdownLogic = kea({
+    actions: {
+        setSelectedItem: (item: SelectedItem) => ({ item }),
+        setSearch: (search: string) => ({ search }),
+        clickSelectedItem: (item: SelectedItem) => ({ item }),
+        setBlockMouseOver: (block: boolean) => ({ block }),
+        onKeyDown: (e) => ({ e }),
+    },
+    reducers: ({ props }) => ({
+        selectedItem: [
+            false,
```

Comment:
we're using `false` as a default for missing values in this file. I think `null` makes much more sense for non-boolean values.